lib/redcord/serializer.rb

```diff
@@ -46,46 +46,68 @@ def decode_attr_value(attribute, val)
       val
     end
 
-    sig { params(attr_key: Symbol, attr_val: T.untyped).returns(T.untyped)}
-    def validate_and_encode_query(attr_key, attr_val)
+    sig { params(attr_key: Symbol, attr_val: T.untyped, index_name: T.nilable(Symbol)).returns(T.untyped)}
+    def validate_and_encode_query(attr_key, attr_val, index_name)
       # Validate that attributes queried for are index attributes
-      if !class_variable_get(:@@index_attributes).include?(attr_key) &&
-         !class_variable_get(:@@range_index_attributes).include?(attr_key)
-        raise(
-          Redcord::AttributeNotIndexed,
-          "#{attr_key} is not an indexed attribute.",
-        )
-      end
-
-      # Validate attribute types for normal index attributes
-      attr_type = get_attr_type(attr_key)
-      if class_variable_get(:@@index_attributes).include?(attr_key)
-        validate_attr_type(attr_val, attr_type)
-      else
-        # Validate attribute types for range index attributes
-        if attr_val.is_a?(Redcord::RangeInterval)
-          validate_attr_type(
-            attr_val.min,
-            T.cast(T.nilable(attr_type), T::Types::Base),
+      if index_name
+        custom_index_attributes = class_variable_get(:@@custom_index_attributes)[index_name] || Array.new  
+        if !custom_index_attributes.include?(attr_key)
+          raise(
+            Redcord::AttributeNotIndexed,
+            "#{attr_key} is not a part of #{index_name} index.",
```

Comment:
what if people provided too many query conditions? do we support joining the result sets from the custom index sets and other index sets?

Reply:
Quick answer - no. My idea was: if you use specify custom index in your query, you are only going to be querying this index.
If you provide attributes in the query that are not part of the specified index - AttributeNotIndexed will be raised.